web/server/codechecker_server/api/report_server.py

```diff
@@ -1562,22 +1562,34 @@ def _setReviewStatus(self, report_id, status, message, session):
                 review_status = ReviewStatus()
                 review_status.bug_hash = report.bug_id
 
-            user = self.__get_username()
-
-            old_status = review_status.status if review_status.status \
-                else review_status_str(ttypes.ReviewStatus.UNREVIEWED)
-            old_msg = review_status.message.decode('utf-8') \
-                if review_status.message else None
+            old_status = review_status.status or \
+                review_status_str(ttypes.ReviewStatus.UNREVIEWED)
+            old_msg = review_status.message or None
+
+            new_status = review_status_str(status)
+            new_user = self.__get_username()
+            new_message = message.encode('utf8') if message else b''
+
+            # Review status is a shared table among runs. When multiple runs
+            # are stored in parallel, there may be a race condition in updating
+            # review status fields. The most common reason of deadlocks is
+            # changing only the date to current date. This condition checks if
+            # something else is also changed other than dates.
+            # TODO: This condition may not be necessary if mass storage is
+            # redesigned so storing transactions run sequentially.
+            if (old_status, review_status.author, old_msg) == \
```

Comment:
I am thinking if only the author is changed we will update the author and dates field of this entry. Wouldn't it be better to keep the original author if nothing (status, message) is changed? This way we would know exactly who wrote the comment.

For example lets assume I wrote a source code comment for a bug and store the results to a run. Next you analyze the same project, and store the results to the same run. It will change the author of my source code comment even if you didn't modify it. Now if someone looks into the source code comment he will see that you wrote this source code comments but its not true.

So my recommendation is to change review status only if the message or the status is changed.

Reply:
I agree. Let's update only if the status or the message changes.

Reply:
Do you agree with our comments? I don't see any update here. You still check whether the author is changed or not here.